### PR TITLE
Assets: can_decrease/increase for destroying asset is not successful

### PR DESCRIPTION
**Pull request by muharem**
Functions `can_decrease` and `can_increase` do not return successful consequence results for assets undergoing destruction; instead, they return the `UnknownAsset` consequence variant.

This update aligns their behavior with similar functions, such as `reducible_balance`, `increase_balance`, `decrease_balance`, and `burn`, which return an `AssetNotLive` error for assets in the process of being destroyed.

### CODE REVIEW

**Review by acatangiu — APPROVED** [review]

File: substrate/frame/assets/src/functions.rs [hunk 1]
```diff
@@ -132,6 +132,9 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			Some(details) => details,
 			None => return DepositConsequence::UnknownAsset,
 		};
+		if details.status == AssetStatus::Destroying {
```
Comment (georgepisaltu) [inline substrate/frame/assets/src/functions.rs:135]: nit: I'd use `matches!` instead of `==`. Same for usage below.

**Review by georgepisaltu — APPROVED** [review]

**Review by joepetrowski — APPROVED** [review]

File: substrate/frame/assets/src/functions.rs (hunk 1 above)
Comment (muharem) [reply substrate/frame/assets/src/functions.rs:135]: what's the benefit? with `==` I follow the style within the function

Comment (georgepisaltu) [reply substrate/frame/assets/src/functions.rs:135]: Pattern matching is the standard way of comparing `enum`s and it doesn't require `PartialEq`/`Eq`, but in this case it's not a big deal.

**Review by liamaharon — APPROVED** [review]

**Comment by bkchr** [comment]
@muharem what is the status of this pr?

**Comment by muharem** [comment]
> @muharem what is the status of this pr?

I did add it to the backlog for audit. But it is ready to be merged.

**Comment by muharem** [comment]
bot merge

**Comment by Polkadot-Forum** [comment]
This pull request has been mentioned on **Polkadot Forum**. There might be relevant details there:

https://forum.polkadot.network/t/discussion-treating-changes-in-dispatchable-function-errors-as-breaking-changes/11222/1

**Pull request merged**